src/app/space/create/deployments/services/deployments.service.ts

```diff
@@ -259,6 +278,47 @@ export class DeploymentsService implements OnDestroy {
       );
   }
 
+  getDeploymentTimeConstrainedStats(spaceId: string, applicationId: string, environmentName: string, elapsedTime: number, endTime: number = Date.now()): Observable<TimeConstrainedStats> {
+    const startTime = endTime - elapsedTime;
+    const stats = this.getTimeConstrainedTimeseriesData(spaceId, applicationId, environmentName, startTime, endTime);
+    // Process CPU Stats
+    let cpuStats: Observable<CoresSeries[]> = stats.map((t: MultiTimeseriesData) => t.cores);
+    const cpuQuota = this.getEnvironmentCpuStat(spaceId, environmentName).map((stat: CpuStat) => stat.quota);
+    cpuStats = Observable.combineLatest(cpuStats, cpuQuota, (cpuStats: CoresSeries[], cpuQuota: number) => {
+      let arr = [];
+      for (let i = 0; i < cpuStats.length; i++) {
+        arr.push({ data: { used: cpuStats[i].value, quota: cpuQuota }, timestamp: cpuStats[i].time });
+      }
+      return arr;
+    });
+    // Process Memory Stats
+    let memStats: Observable<MemorySeries[]> = stats.map((t: MultiTimeseriesData) => t.memory);
+    const memQuota = this.getEnvironment(spaceId, environmentName).map((env: EnvironmentStat) => env.quota.memory.quota);
+    memStats = Observable.combineLatest(memStats, memQuota, (memStats: MemorySeries[], memQuota: number) => {
+      let arr = [];
+      for (let i = 0; i < memStats.length; i++) {
+        arr.push({ data: new ScaledMemoryStat(memStats[i].value, memQuota), timestamp: memStats[i].time });
+      }
+      return arr;
+    });
+    // Process Network Stats
+    const net_txStats: Observable<NetworkSentSeries[]> = stats.map((t: MultiTimeseriesData) => t.net_tx);
+    const net_rxStats: Observable<NetworkReceivedSeries[]> = stats.map((t: MultiTimeseriesData) => t.net_rx);
+    let networkStats: Observable<NetworkStat[]> = Observable.combineLatest(net_txStats, net_rxStats).map((n) => {
+      let arr = [];
+      for (let i = 0; i < n[0].length; i++) {
+        arr.push({ data: { sent: new ScaledNetworkStat(n[0][i].value), received: new ScaledNetworkStat(n[1][i].value) }, timestamp: n[0][i].time });
+      }
+      return arr;
+    });
+    return Observable.combineLatest(cpuStats, memStats, networkStats, (
+        cpuStats: { data: { used: number, quota: number }, timestamp: number }[],
+        memStats: { data: ScaledMemoryStat, timestamp: number }[],
+        networkStats: { data: { sent: ScaledNetworkStat, received: ScaledNetworkStat }, timestamp: number }[]
+      ) => ({ cpu: cpuStats, memory: memStats, network: networkStats })
```

Comment:
Here's where `TimeConstrainedCpuStats` would be a nice type to reference for these lambda params.